mesonbuild/backend/ninjabackend.py

```diff
@@ -2387,6 +2387,73 @@ def get_link_whole_args(self, linker, target):
         target_args = self.build_target_link_arguments(linker, target.link_whole_targets)
         return linker.get_link_whole_for(target_args) if len(target_args) else []
 
+    def guess_library_absolute_path(self, libname, search_dirs, prefixes, suffixes):
+        for directory in search_dirs:
+            for suffix in suffixes:
+                for prefix in prefixes:
+                    trial = os.path.join(directory, prefix + libname + '.' + suffix)
+                    if os.path.isfile(trial):
+                        return trial
+
+    def guess_external_link_dependencies(self, linker, target, commands, internal):
+        # This is a work around for missing dependency file generation in the linker
+        # https://sourceware.org/bugzilla/show_bug.cgi?id=22843
+        search_dirs = []
+        libs = []
+        absolute_libs = []
+
+        build_dir = self.environment.get_build_dir()
+
+        it = iter(commands)
+        try:
+            while True:
+                item = next(it)
```

Comment:
Why are you converting the list to an iterator and then looping over it anyway?

Reply:
Sometimes the code needs 2 items. i.e. '-L', '/some/path'. I think having an iterator to pull from is easier than doing the more like a state machine.

Reply:
... actually maybe we should warn when there's no argument after an empty `-L`, so we don't want to silently `pass` in that case. Or maybe an error, because this will cause weird bugs where the next argument gets incorrectly slurped as a library search directory.

Reply:
I wrote this code with the assumption that it should never break a build that worked before, because this is a bit of a second guessing game with the linker. The linker is the actual point of truth.

Not sure what mesons policy on warning is. But i could add a working if the commands end with an unmatches `-L` or `-l`

Reply:
I've added a warning and refactored the looping code. I didn't know that a iterator is valid for a `for` loop as well as an iterable, but it seems to work.